analyzers/src/SonarAnalyzer.Common/Rules/Utilities/SymbolReferenceAnalyzerBase.cs

```diff
@@ -42,22 +41,21 @@ public abstract class SymbolReferenceAnalyzerBase<TSyntaxKind> : UtilityAnalyzer
 
         protected sealed override SymbolReferenceInfo CreateMessage(SyntaxTree syntaxTree, SemanticModel semanticModel)
         {
-            var symbolReferenceInfo = new SymbolReferenceInfo { FilePath = syntaxTree.FilePath };
+            var filePath = GetFilePath(syntaxTree);
+            var symbolReferenceInfo = new SymbolReferenceInfo { FilePath = filePath };
             var references = GetReferences(syntaxTree.GetRoot(), semanticModel);
-
             foreach (var symbol in references.Keys)
             {
-                if (GetSymbolReference(references[symbol], syntaxTree) is { } reference)
+                if (GetSymbolReference(references[symbol], filePath) is { } reference)
                 {
                     symbolReferenceInfo.Reference.Add(reference);
                 }
             }
-
             return symbolReferenceInfo;
         }
 
-        protected override bool ShouldGenerateMetrics(SyntaxTree tree) =>
-            base.ShouldGenerateMetrics(tree)
+        protected override bool ShouldGenerateMetrics(SyntaxTree tree, Compilation compilation) =>
+            base.ShouldGenerateMetrics(tree, compilation)
```

Comment:
I like this more on two lines (because `base` call is somewhat special), but I think it should go on a single line.

Reply:
Aren't most of our conditions on separate lines? The code is easier to read and the coverage is much easier to see on SC.